healpy/__init__.py

```diff
@@ -107,16 +105,9 @@
 from ._hotspots import hotspots_healpy as hotspots
 from ._line_integral_convolution import line_integral_convolution
 
+# Default null handler prevents printing all log messages if the user doesn't
+# configure `logging`
+# See https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library
+import logging
 
-def disable_warnings():
-    """Disable all healpy warnings messages for the current session
-
-    Warnings from individual functions can be disabled setting
-    ``verbose=False``.
-    Warnings can be re-enabled calling ``hp.enable_warnings()``.
-    """
-    warnings.filterwarnings(action="ignore", module="healpy")
-
-
-def enable_warnings():
-    warnings.simplefilter("always")
\ No newline at end of file
+logging.getLogger("healpy").addHandler(logging.NullHandler())
```

Comment:
It would be a good idea to see if there is any benefit to directly tapping into Astropy's warning system.

Reply:
I think `logging` only except for deprecation is fine.